sympy/core/core.py

```diff
@@ -42,6 +44,31 @@
 ]
 
 
+def old_compare(x: type, y: type) -> int:
```

Comment:
Before this was part of a method, so `x` would always be `issubclass(x, Basic)`. This is no longer the case. Could this lead to any problems?

Reply:
I don't worry about this too much be cause the comparison is not used often and is symmetric. If there are some problems crop up, this can be made more robust

Reply:
The only place that calls this is `Basic.compare` so `x` will always be a subclass of `Basic`. It could be problematic though if users used this as a public function.

Having this function here in `core.py` creates a circular import because it has to import `Basic`. I suggest moving this function to `basic.py` and making it private like `_old_compare`. It shouldn't be public if it isn't really something that we want to support in future. On the other hand if it should not be private then I would call it something like `compare_classes` rather than `old_compare`.

Reply:
Apart from the name/location of this function I think the PR looks good.